### DISCUSSION TITLE
Connection to server failed context deadline exceeded

### DISCUSSION THREAD

**Discussion by aHyperGG** [BishopFox/sliver#1320] [Q&A] [answered]
Hello i changed my netword adaptater and sliver does not work anymore

**Comment by moloch--** [comment]
This error typically means the TCP port is being block by a firewall or some similar type of networking problem.

**Reply by aHyperGG** [reply]
thank you very much but how can i fix it?

**Reply by moloch--** [reply]
Its not a problem with Sliver, you'll need to ensure your system/server can route TCP traffic to the gRPC listener port.

**Comment by manasmbellani** [comment]
Hi, The reason this error occurred for me was that I had to manually start the sliver server before the client is started.
So I started sliver server first and I was able to get sliver client working. 

```
./sliver-server
./sliver-client_linux
```

**Reply by aHyperGG** [reply]
thank you very much!

**Comment by deeexcee-io** [comment, marked as answer]
Had the same issue this morning. Running on Ubuntu. Try starting the sliver service 
```bash
sudo systemctl start sliver.service
```

**Reply by serWazito0** [reply]
This one solved the same error with me.

**Comment by xyakaix** [comment]
I have tried all the proposed solutions above, but still no luck. It used to work on AWS several months ago. I suspect VPC changes of AWS are the reason, but still haven't found a solution. Anyone have similar issue?

**Comment by l0n3m4n** [comment]
I believe the problem lies in the local bin. ive tried multiple methods, but it hasn’t worked. so decide to try using wget with both the client and server packages instead of using a one-liner and  stored in home directory then it work.

**Reply by moloch--** [reply]
The error simply indicates the client was unable to connect to the server, which may have any number of causes. The most common though is always going to be networking issues (Firewalls, etc.), the server daemon simply isn't running, or the TCP port is incorrect.

**Comment by NatRidgard** [comment]
I tried disabling UFW but what worked for me was running curl https:/sliver.sh/install|sudo bash
As I noticed it in https://www.youtube.com/watch?v=gvOPFY13nKc

Cheers folks

**Comment by grugnoymeme** [comment]
this could be caused by moving sliver-client and sliver-servet into /usr/local/bin instead of the default /root position.
check it typing "which sliver", if u have it in /root just start the sliver service systemctl (sudo systemctl start sliver.service) and also the sliver service (sudo service sliver start).
Insead, if you moved the executables in /usr/local/bin just edit the systemd's sliver's file: (sudo nano /etc/systemd/system/sliver.service) and replace the "ExecStart=/root/sliver-server daemon" row updating it with "ExecStart=/usr/local/bin/sliver-server daemon".
Then restart the daemon and the service:
sudo systemctl daemon-reexec
sudo systemctl daemon-reload
sudo systemctl enable sliver.service
sudo systemctl restart sliver.service
Now should be fine.
happy hack
